### ISSUE TITLE
plot(adp,which=80) produces confusing error

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1562]
This came up in test code for #1561.

The problem with the second arg being `""` may be something wider, related to changes for #1554. But, in any case, a `plot()` call is at a high enough level that the user ought to get something that will mean more to them.

```R
plot(adp, which= 80 )

 Error: object[["vv", ""]]: second arg must be "data" or "metadata" 
8.
stop("object[[\"", i, "\", \"", j, "\"]]: second arg must be \"data\" or \"metadata\"", 
    call. = FALSE) at AllClass.R#402
7.
.nextMethod(x = x, i = i, j = j) 
6.
callNextMethod() at adp.R#817
5.
x[["vv", j]] 
4.
x[["vv", j]] at adp.R#1954
3.
.local(x, ...) 
2.
plot(adp, which = which) 
1.
plot(adp, which = which) 
```

**Comment by dankelley** [comment]
Hm... I sort of thought this might happen when I was changing the base-level code for `[[` for issue #1554.

In this particular case, the problem stems from the following in `R/adp.R` at line 1952:
```R
                      } else if (which[w] == 80) {
                          ## vertical beam velocity
                          z <- x[["vv", j]]
                          if (!is.null(z)) {
```
When `[[` is called, it looks for an adp-level specialization for `"vv"` and, finding none, falls back to the base-level `[[`. And, at that level, for the new code for #1554, we only permit `j` to be `NULL`, or `data`, or `metadata`. 

The solution is to make `[[,adp-method` notice when first argument is `"vv"`, and handle that.  I don't want this handled at the low-level `[[` because it might be specialized (e.g. for the `landsat` object, we do tricks of assembling two bytes into an integer, etc ... stuff like that belongs in the specialized object, not in a wild mass of `if` statements in the low-level version of `[[`).

PS. in this case, there is no `vv` field, and once I fix up `[[`, the `plot,adp-method` call will notice that `z` is `NULL`, and will spit out a warning, without plotting.

**Comment by dankelley** [comment]
I am enlarging the test suite to handle all documented values of `which`, and, as part of that, I am testing for specific warning messages. This has revealed that some of the messages are worded differently from others (e.g. "which=X not allowed because of Y", as opposed to "Y, so cannot plot with which=X" sort of wordings), and I do not like that, so I am making the warnings have a more common format.

But now I'm thinking ... why the heck does spit warnings, instead of errors?  In normal R, if I do e.g. `plot(NULL)`, it doesn't give a warning; rather it gives an error.

I don't want users thinking things are okay when they are not OK. That's a path for confusion!

Therefore, I am going to convert those warnings into errors.

I expect this work to be done by noon today.

**Comment by dankelley** [comment]
This is fixed in the "develop" branch commit 8bd03361b5b29fec68038fdff2a5eeea891c463f. I have also added tests of every single documented `which` value (see `tests/testthat/test_adp.R` and `tests/testthat/test_ad2cp.R`).
